### PR TITLE
Youtube front end

### PR DESCRIPTION
**Pull request by faroligo**
fixes #77 
@meadsteve @Harxy Can be tested here http://faroligo-jschat.herokuapp.com/rooms/dev

### CODE REVIEW

**Comment by meadsteve** [comment]
Firstly just checked out the demo. Looks awesome. And the r b Rebecca black is an excellent touch.

File: www/scripts/main.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
+﻿var userNameField = $("#name");
```
Comment (meadsteve) [inline www/scripts/main.js]: I think there's a small chance this code could be executed before the dom is loaded which will cause some annoying bugs. I think it's worth moving all the `$(...)` inside a `$(function(){...})` so we can be assured that the dom is ready.

File: www/scripts/CHABBLE.YouTubeClient.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    var playState;
+    var readyCallback;
+    var finishedCallback;
+    var readyToPlay;
+
+    function playYouTubeVideo(videoId, startTime) {
+        var currentTime = youTubePlayer.getCurrentTime();
+        var outBy = Math.abs(currentTime - startTime);
+
+        if ( (playState === 1 && outBy < 10) || playState === 3)
```
Comment (meadsteve) [inline www/scripts/CHABBLE.YouTubeClient.js]: Not essential but could we make the play state stuff a bit easier to read by defining an object with the values mapped on to something readable at the start. so something like:

``` js
var playState;
var playStates = {
   "not-started": 0
   "playing": 1,
   "paused": 3
};
```

This if statement could then read:

``` js
if ( (playState === playStates.playing && outBy < 10) || playState === playStates.paused)
```

n.b. I don't know what the states actually are so the above is an example I made up randomly.

**Comment by faroligo** [comment]
@meadsteve I was wondering how to recreate an enum in JS!  So we are pretty close on all three bits being done (I think @Harxy is almost done).   Who is going to integrate?

**Comment by meadsteve** [comment]
@faroligo yeah it's a bit of a poor enum approach but think it's the best you can do pre-es6. 

Regarding integrating I figure we get all the bits merged then see who has the time? I'm gonna do a small piece now to make sure my queue is sending messages to the channel this PR is listening ti.

**Pull request merged**